src/cb_admin/pages/Visitor.js

```diff
@@ -139,9 +144,20 @@ export default class VisitorProfile extends React.Component {
   };
 
   onClickResend = () => {
+    this.setState({ resendQrCodeState: resendQrCodeState.PENDING });
     Visitors.sendQrCode({ id: this.state.id })
-      .then(() => this.setState({ hasResent: true }))
-      .catch(err => redirectOnError(this.props.history.push, err));
+      .then(() => this.setState({ resendQrCodeState: resendQrCodeState.SUCCESS }))
+      .catch((err) => {
+        if (ErrorUtils.errorStatusEquals(err, 400)) {
+          this.setState({
+            resendQrCodeState: resendQrCodeState.ERROR,
+            errors: err.response.data.error,
```

Comment:
The `errors` state fragment is an object whose keys are the names of input fields/elements (see how its used in passing values to `LabelledInput` below). Setting `errors.resendBtn` or similar is more appropriate.